src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -322,7 +322,7 @@ void checkAssignable(DiagnosticPosition pos, VarSymbol v, JCTree base, Env<AttrC
                    withfield operator -This does not result in mutation of final fields; the code generator
                    would implement `copy on write' semantics via the opcode `withfield'.
                 */
-                if (env.info.inWithField && v.getKind() == ElementKind.FIELD && (v.flags() & STATIC) == 0 && types.isPrimitiveClass(v.owner.type)) {
+                if (env.info.inWithField && v.getKind() == ElementKind.FIELD && (v.flags() & STATIC) == 0 && v.owner.isPrimitiveClass()) {
                     if (env.enclClass.sym.outermostClass() == v.owner.outermostClass())
```

Comment:
This is an example of a code change you will see recur several times: Earlier when we had only value default primitive classes to check if a field is a field of a primitive class it was alright to ask
```
types.isPrimitiveClass(v.owner.type)
```
With ref-default classes entering the picture if v.owner is a ref-default primitive class then v.owner.type will not
answer true for the above check. (For a ref-default primitive class Optional.val, Optional.type is 'L' type not Q type) and we need to check
```
owner.isPrimitiveClass()
```
which will answer true for both val-default and ref-default primitive classes.

Reply:
Fair points. 

Since we already have com.sun.tools.javac.code.Type#isPrimitive it is useful to retain to "Class" somewhere in the new APIs name although perhaps not in the present form of Type#isPrimitiveClass()

Perhaps the old method should be renamed to isBasicPrimitive() and the new one should be isPrimitiveClassType()

I will carry over these comments to JDK-8268734 (Various internal Type/Symbol APIs need rationalization/adjustment)